src/command_modules/azure-cli-network/azure/cli/command_modules/network/custom.py

```diff
@@ -1288,6 +1311,11 @@ def set_nic_ip_config(instance, parent, ip_config_name, subnet=None,
     elif load_balancer_inbound_nat_rule_ids is not None:
         instance.load_balancer_inbound_nat_rules = load_balancer_inbound_nat_rule_ids
 
+    if application_security_groups == ['']:
```

Comment:
So, I don't do the API check on update, because the update statements are always standalone.
```Python
if foo:
  instance.foo = foo
if bar:
  instance.bar = bar
```

The problem occurs during creates:
```Python
thing = Thing(foo=foo)
if bar:
  thing.bar = bar
```
If bar depends on an API version, it will break if someone refactors this to:
```Python
thing = Python(foo=foo, bar=bar)
```

This can't happen in the update scenario, so I don't use the API checks in any update scenarios.